src/main/java/io/quarkus/search/app/indexing/IndexingService.java

```diff
@@ -101,6 +85,25 @@ void indexOnStartup(@Observes StartupEvent ev) {
                         () -> Log.infof("Search backend is not ready yet, waiting...")))
                 .chain(() -> Uni.createFrom()
                         .item(() -> {
+                            if (IndexingConfig.OnStartup.When.INDEXES_EMPTY.equals(indexingConfig.onStartup().when())) {
+                                try {
+                                    long documentCount = QuarkusTransaction.requiringNew().call(
+                                            () -> searchSession.search(Object.class)
+                                                    .where(f -> f.matchAll())
+                                                    .fetchTotalHitCount());
+                                    if (documentCount >= 0L) {
```

Comment:
Can we have a case when the docs count is 0, and we do not want to reindex things?

Reply:
Good catch, I'll fix that. Thanks.